### PR TITLE
Update helm to 3.5.3 (also update architect-orb for ci)

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR updates helm to 3.5.3 to (hopefully) unblock giantswarm/apptestctl#104

The old version had k8s.io/kubernetes 1.13 as a dependency.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,32 +11,31 @@ require (
...
-	helm.sh/helm/v3 v3.4.2
-	k8s.io/api v0.19.4
-	k8s.io/apimachinery v0.19.4
-	k8s.io/client-go v0.19.4
-	k8s.io/utils v0.0.0-20200912215256-4140de9c8800 // indirect
-	rsc.io/letsencrypt v0.0.3 // indirect
+	helm.sh/helm/v3 v3.5.3
+	k8s.io/api v0.20.2
+	k8s.io/apimachinery v0.20.2
+	k8s.io/client-go v0.20.2
```
Comment (rossf7) [inline go.mod:25]: The problem is the CAPI forks are still using v0.18.6
https://github.com/giantswarm/cluster-api/blob/v0.3.13-gs/go.mod#L40

Can you try vendoring this in chart-operator? If it compiles and the tests pass this should be fine to merge.

Comment (ubergesundheit) [reply go.mod:25]: Seems to work https://github.com/giantswarm/chart-operator/pull/708

Comment (rossf7) [reply go.mod:25]: Nice, thanks for checking.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**